src/sema/types.rs

```diff
@@ -1025,18 +1025,18 @@ impl Type {
             Type::InternalFunction { .. } => BigInt::from(ns.target.ptr_size()),
             Type::ExternalFunction { .. } => {
                 // Address and selector
-                Type::Address(false).size_of(ns) + Type::Uint(32).size_of(ns)
+                Type::Address(false).storage_size_of(ns) + Type::Uint(32).storage_size_of(ns)
             }
             Type::Unresolved | Type::Mapping(..) => BigInt::zero(),
-            Type::Ref(ty) | Type::StorageRef(_, ty) => ty.size_of(ns),
-            Type::UserType(no) => ns.user_types[*no].ty.size_of(ns),
+            Type::Ref(ty) | Type::StorageRef(_, ty) => ty.storage_size_of(ns),
+            Type::UserType(no) => ns.user_types[*no].ty.storage_size_of(ns),
             _ => unimplemented!("sizeof on {:?}", self),
         }
     }
 
     /// Does this type fit into memory
     pub fn fits_in_memory(&self, ns: &Namespace) -> bool {
```

Comment:
@seanyoung I'm not sure here. Does this function require any changes?

Reply:
I'm going to create a function to estimate the size of each type in memory. I use this function in my Borsh encoding algorithm too.